drivers/spi/spi_ll_stm32.c

```diff
@@ -73,71 +73,123 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 	return 0;
 }
 
-static inline u16_t spi_stm32_next_tx(struct spi_stm32_data *data)
+static inline void spi_stm32_shift_m8(SPI_TypeDef *spi,
+				      struct spi_stm32_data *data)
 {
+	struct spi_context *ctx = &(data->ctx);
 	u16_t tx_frame = SPI_STM32_TX_NOP;
-
-	if (spi_context_tx_buf_on(&data->ctx)) {
-		if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
-			tx_frame = UNALIGNED_GET((u8_t *)(data->ctx.tx_buf));
-		} else {
-			tx_frame = UNALIGNED_GET((u16_t *)(data->ctx.tx_buf));
-		}
+	u16_t rx_frame;
+	u8_t frame_sz;
+
+	if (spi_context_tx_buf_on(ctx) && spi_context_rx_buf_on(ctx)) {
+		frame_sz = MIN(ctx->tx_len, ctx->rx_len);
+	} else if (spi_context_tx_buf_on(ctx)) {
+		frame_sz = ctx->tx_len;
+	} else if (spi_context_rx_buf_on(ctx)) {
+		frame_sz = ctx->rx_len;
+	} else {
+		/* Should never get here, but place case for it anyways */
+		return;
 	}
 
-	return tx_frame;
-}
+#if defined(CONFIG_SPI_STM32_HAS_FIFO)
```

Comment:
you could use IS_ENABLED() instead to avoid #ifdef

Reply:
If I'm correct, this won't work as `ll_func_set_fifo_threshold_xbit` functions are not defined if CONFIG_SPI_STM32_HAS_FIFO not defined.

Reply:
indeed, I missed the fact the new inline function was within a former ifdef.